src/graphql/Offer/permissions.ts

```diff
@@ -0,0 +1,15 @@
+import { and } from "graphql-shield";
+import { isAuthenticated, isAuthorized } from "../rules";
+
+export const offerPermissions = {
+  Query: {
+    getOffers: isAuthenticated,
+    getOfferByUuid: isAuthenticated
+  },
+  Mutation: {
+    saveOffer: isAuthenticated
+  },
+  Offer: {
+    hasApplied: and(isAuthenticated, isAuthorized)
```

Comment:
siempre isAuthorized va a estar en un and con isAuthenticated, lo haría incluirlo dentro suyo

Reply:
terminaria repitiendo codigo si hago esto. La idea del middleware es manejar las reglas asi. Pensa que hay queries de User que solo necesitan isAuthenticated y hay queries de Applicant, Company y Admin que necesitan ademas las respectivas